sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -198,6 +200,10 @@ case class CatalogTable(
       locationUri, inputFormat, outputFormat, serde, compressed, properties))
   }
 
+  def withStats(cboStatsEnabled: Boolean): CatalogTable = {
```

Comment:
Do we need to get rid of the `CatalogStatistics` if the config is off? Actually I think you can decide whether to use it or not when doing estimation later, depending on this config. It seems no harm to always attach this info to `CatalogTable`.